src/routes/transactions/helpers/swap-order.helper.ts

```diff
@@ -138,6 +149,46 @@ export class SwapOrderHelper {
     if (!transaction.data) return false;
     return this.setPreSignatureDecoder.isSetPreSignature(transaction.data);
   }
+
+  /**
+   * Retrieves a token object based on the provided Ethereum chain ID and token address.
+   * If the specified address is the placeholder for the native currency of the chain,
+   * it fetches the chain's native currency details from the {@link IChainsRepository}.
+   * Otherwise, it fetches the token details from the {@link ITokenRepository}.
+   *
+   * @param args An object containing:
+   *   - `chainId`: A string representing the ID of the blockchain chain.
+   *   - `address`: A string representing the Ethereum address of the token, prefixed with '0x'.
+   * @returns {Promise<Token>} A promise that resolves to a Token object containing the details
+   * of either the native currency or the specified token.
+   * @throws {Error} Throws an error if the token data cannot be retrieved.
+   * @private
+   * @async
+   */
+  private async getToken(args: {
+    chainId: string;
+    address: `0x${string}`;
+  }): Promise<Token> {
+    if (args.address === SwapOrderHelper.NATIVE_CURRENCY_ADDRESS) {
```

Comment:
Are we sure that `args.address` with _always_ be checksummed? It would be a bit safer to either checksum `args.address` or compare them in lowercase. What do you think?

Reply:
Auto-merge merged the commit before I was able to tackle this. I will open a follow up PR.

Reply:
Tackled it here: https://github.com/safe-global/safe-client-gateway/pull/1493